evalml/automl/automl_algorithm/evalml_algorithm.py

```diff
@@ -0,0 +1,367 @@
+
+    At this point we have a single pipeline candidate for preprocessing, feature engineering and feature selection
+
+    5. Pipelines with preprocessing components:
+        a. scan estimators (our current batch 1).
+        b. Then run ensembling
+
+    Fast mode ends here. Begin long mode.
+
+    6. Run some random pipelines:
+        a. Choose top 3 estimators. Generate 50 random parameter sets. Run all 150 in one batch
+    7. Run ensembling
+    8. Repeat these indefinitely until stopping criterion is met:
+        a. For each of the previous top 3 estimators, sample 10 parameters from the tuner. Run all 30 in one batch
+        b. Run ensembling
+    """
+
+    def __init__(
+        self,
+        X,
+        y,
+        problem_type,
+        _sampler_name,
+        tuner_class=None,
+        random_seed=0,
+        pipeline_params=None,
+        custom_hyperparameters=None,
+        n_jobs=-1,
+        number_features=None,
+        text_in_ensembling=None,
+    ):
+        """
+        Arguments:
+            X (pd.DataFrame): Training data
+            y (pd.Series): Target data
+            problem_type (ProblemType): Problem type associated with training data
+            _sampler_name (BaseSampler): Sampler to use for preprocessing
+            tuner_class (class): A subclass of Tuner, to be used to find parameters for each pipeline. The default of None indicates the SKOptTuner will be used.
+            random_seed (int): Seed for the random number generator. Defaults to 0.
+            n_jobs (int or None): Non-negative integer describing level of parallelism used for pipelines. Defaults to -1.
+            pipeline_params (dict or None): Pipeline-level parameters that should be passed to the proposed pipelines. Defaults to None.
+            custom_hyperparameters (dict or None): Custom hyperparameter ranges specified for pipelines to iterate over. Defaults to None.
+            text_in_ensembling (boolean): If True and ensembling is True, then n_jobs will be set to 1 to avoid downstream sklearn stacking issues related to nltk. Defaults to None.
+        """
+
+        super().__init__(
+            allowed_pipelines=[],
+            custom_hyperparameters=custom_hyperparameters,
+            max_iterations=None,
+            tuner_class=None,
+            random_seed=random_seed,
+        )
+
+        self.X = X
+        self.y = y
+        self.problem_type = problem_type
+        self._sampler_name = _sampler_name
+
+        self.n_jobs = n_jobs
+        self.number_features = number_features
+        self._best_pipeline_info = {}
+        self.text_in_ensembling = text_in_ensembling
+        self._pipeline_params = pipeline_params or {}
+        self._custom_hyperparameters = custom_hyperparameters or {}
+        self._selected_cols = None
+        self._top_n_pipelines = None
+        if custom_hyperparameters and not isinstance(custom_hyperparameters, dict):
```

Comment:
There's some duplicate logic from `IterativeAlgorithm` that we can refactor into a base class later on.